src/core/network/webhook.ts

```diff
@@ -8,43 +8,39 @@ export default function (
   updateHandler: (update: Update, res: http.ServerResponse) => Promise<void>,
   errorHandler: (err: SyntaxError) => void
 ) {
-  return (
+  return async (
     req: http.IncomingMessage,
     res: http.ServerResponse,
-    next?: () => void
-  ): void => {
-    debug('Incoming request', req.method, req.url)
-    if (req.method !== 'POST' || req.url !== hookPath) {
-      if (typeof next === 'function') {
-        return next()
-      }
+    next = (): void => {
       res.statusCode = 403
       return res.end()
     }
+  ): Promise<void> => {
+    debug('Incoming request', req.method, req.url)
+    if (req.method !== 'POST' || req.url !== hookPath) {
+      return next()
+    }
     let body = ''
-    req.on('data', (chunk: string) => {
-      body += chunk.toString()
-    })
-    req.on('end', () => {
-      let update: Update
-      try {
-        update = JSON.parse(body)
-      } catch (error) {
-        res.writeHead(415)
+    for await (const chunk of req) {
+      body += (chunk as Buffer).toString()
```

Comment:
I hate this cast, is `chunk` being `any` a bug of `@types/node`?

Reply:
i do this cast because i got an error about + operator only is usable if both are string

Reply:
```suggestion
    for await (const chunk of req) {
      body += String(chunk)
```

Reply:
I'd rather perform the cast consistently with
https://github.com/telegraf/telegraf/blob/056dca05c53053d656de538365ec057dc677d806/src/core/network/client.ts#L154

Reply:
I'd argue `String` is clearer.

Reply:
i used as Buffer because its a typescript code and doesn't change the js output

but using `${body}` or String(body) changes js code